ibis/expr/api.py

```diff
@@ -1869,8 +1867,35 @@ def _timestamp_time(arg):
 # ---------------------------------------------------------------------
 # Time API
 
+def between_time(arg, lower, upper, timezone=None):
+    """
+    Check if the input expr falls between the lower/upper bounds
+    passed. Bounds are inclusive. All arguments must be comparable.
+
+    Parameters
+    ----------
+    lower : str, datetime.time
+    upper : str, datetime.time
+    timezone : str, timezone, default None
+
+    Returns
+    -------
+    is_between : BooleanValue
+    """
+
+    # validate that we are part of a time() column
+    bt = ir.find_base_table(arg)
+    if bt is None:
+        raise TypeError("cannot operate on a non-time columnn")
```

Comment:
Since we're special casing expressions that look like this:

```python
<some column expression>.time().between(..., ...)
```

 we should do something like this:

```python
if isinstance(arg.op(), _ops.Time):
    arg = arg.op().args[0].cast(dt.Timestamp(timezone=timezone))
    return _ops.BetweenTime(arg, lower, upper)
else:
    return _ops.Between(arg, lower, upper)
```